### DISCUSSION TITLE
Error : Call to a member function autoGeneratesTitles() on null

### DISCUSSION THREAD

**Discussion by Nathan269** [statamic/cms#8435] [Troubleshooting] [answered]
Hi folks,

I've just upgraded Statamic from version 3.4 to 4.0 and I'm trying to clean up the test suite. I was initially getting this error when I tried to create an entry within my test:

LogicException : Cannot set date on non-dated collection entry.

After checking the docs, I was able to resolve this issue, but when I rerun my tests I now get this error:

Error : Call to a member function autoGeneratesTitles() on null

This is an example of one of the tests where I get the above error:

          public function it_can_fetch_new_embed_code_on_save()
          {
               $entry = Entry::make()
                   ->slug('my_slug')
                   ->data($this->getEntryData());
  
              $entry->collection(Collection::findByHandle('news'));
              $entry->save();
  
              $this->assertNotEmpty($entry->data()['bard'][16]['attrs']['values']['embed']);
              $this->assertNotEmpty($entry->data()['bard'][22]['attrs']['values']['embed']);
              $this->assertNotEmpty($entry->data()['bard'][29]['attrs']['values']['embed']);
              $this->assertNotEmpty($entry->data()['bard'][33]['attrs']['values']['embed']);
              $this->assertNotEmpty($entry->data()['bard'][35]['attrs']['values']['embed']);
              $this->assertNotEmpty($entry->data()['bard'][37]['attrs']['values']['embed']);
         }

    
Has anyone else experienced this error yet?

**Comment by ryanmitchell** [comment]
Did you try setting the collection before you set data()

**Reply by Nathan269** [reply]
I did have the previous error working earlier but now I'm getting it again:

LogicException : Cannot set date on non-dated collection entry.

I have this:

$entry = Entry::make()
            ->slug('my_slug')
            ->data($this->getEntryData())
            ->collection(Collection::findByHandle('news'))
            ->date(now());

$entry->save();

**Reply by duncanmcclean** [reply]
You can't do `->date()` if the collection isn't configured as a "Dated collection":

![image](https://github.com/statamic/cms/assets/19637309/82ba5524-9e61-46f8-aa55-63f62c3f4dd7)

**Reply by Nathan269** [reply]
Hi Duncan, I already have that set to true for my "news" collection.

**Reply by duncanmcclean** [reply]
Oh weird, it'll only throw that error if the collection doesn't have it enabled, see:

https://github.com/statamic/cms/blob/ee77de74f7a43e7f874ae74bd50f5917e3e27071/src/Entries/Entry.php#L469

**Reply by Nathan269** [reply]
Yeah that's odd. At the top of my json for my settings for the "news" collection, I have the following:

"dated": true,

**Reply by duncanmcclean** [reply]
What if you `dd(Collection::findByHandle('news'))` in your test - is it dated there?

**Reply by Nathan269** [reply]
Ah it's returning null. This is obviously why it keeps failing. I'll report back when I have that part sorted.

**Reply by Nathan269** [reply]
The only way I can get a return for dd(Collection::findByHandle('news')) is if I do the following:

$entry = Entry::make()
            ->slug('my_slug')
            ->data($this->getEntryData());

dd($entry->collection(Collection::findByHandle('news')));
$entry->save();
        
If I try this:

$entry = Entry::make()
            ->slug('my_slug')
            ->data($this->getEntryData())
            ->collection(Collection::findByHandle('news'))
            ->date(now());

$entry->save();
        
I continue to get this error:

LogicException : Cannot set date on non-dated collection entry.

This is what the docs say for upgrading to statamic version 4:

When using the date method on an entry, an exception will be thrown if the entry is not in a dated collection.

Because of this, if you are creating an entry instance, you should set the collection before the date.

**Reply by Nathan269** [reply, marked as answer]
Although my test setup was ok, my database seeders were causing the issues. It was just a matter making sure the collection was created and then set for the entry before the date was set.
